pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -250,13 +250,33 @@ public PlanNode makeSegmentPlanNode(IndexSegment indexSegment, QueryContext quer
         return new AggregationPlanNode(indexSegment, queryContext);
       }
     } else if (QueryContextUtils.isSelectionQuery(queryContext)) {
+      validateSelectionOrderByExpressions(indexSegment, queryContext);
```

Comment:
Is this the earliest stage that this check can be done? We should try to do such semantic checks as soon as possible (during or right after parsing) so that the query can be failed as quickly as possible without consuming any extra resources. There are a bunch of checks in BrokerRequestHandler, so I am wondering if this check can be done there instead of in server side plan generation?